### PR TITLE
Add complex example of fileglob with loop

### LINKED ISSUE #48788: loop w/ fileglob does not accept lists
##### SUMMARY

When trying to replace all `with_whatever:` with `loop: q()` I ran into the quirk that `q("fileglob", …)` does not accept lists and fails with
```
An unhandled exception occurred while running the lookup plugin 'fileglob'. Error was a <type 'exceptions.AttributeError'>, original message: 'list' object has no attribute 'rfind'"
```
Maybe I'm missing something. In that case, please be so kind and give me a working example, thanks!

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME

fileglob

##### ANSIBLE VERSION

```paste below
ansible 2.7.1
python version = 2.7.15 (default, Oct 15 2018, 15:24:06) [GCC 8.1.1 20180712 (Red Hat 8.1.1-5)]
```

##### CONFIGURATION

```paste below
n/a
```

##### OS / ENVIRONMENT

Fedora 28, x86_64

##### STEPS TO REPRODUCE

```bash
mkdir -p files/{foo,bar}
touch files/{foo,bar}/{baz,yatta}.conf
```

```yaml
---
- hosts: localhost
  gather_facts: no
  vars:
    mygroups:
      - foo
      - bar
  tasks:
    - debug: var=item
      with_fileglob: '{{mygroups|map("regex_replace", "^(.*)$", "files/\1/*.conf")|list}}'

    - debug: var=item
      loop: '{{q("fileglob",
        mygroups|map("regex_replace", "^(.*)$", "files/\1/*.conf")|list
        )}}'
```

##### EXPECTED RESULTS

```
TASK [debug] ******************************************************************************                                                                                          
ok: [localhost] => (item=/tmp/ansible/files/foo/yatta.conf) => {                                                                                                      
    "item": "/tmp/ansible/files/foo/yatta.conf"                                                                                                                       
}
ok: [localhost] => (item=/tmp/ansible/files/foo/baz.conf) => {
    "item": "/tmp/ansible/files/foo/baz.conf"
}
ok: [localhost] => (item=/tmp/ansible/files/bar/yatta.conf) => {
    "item": "/tmp/ansible/files/bar/yatta.conf"
}
ok: [localhost] => (item=/tmp/ansible/files/bar/baz.conf) => {
    "item": "/tmp/ansible/files/bar/baz.conf"
}
TASK [debug] ******************************************************************************                                                                                          
ok: [localhost] => (item=/tmp/ansible/files/foo/yatta.conf) => {                                                                                                      
    "item": "/tmp/ansible/files/foo/yatta.conf"                                                                                                                       
}
ok: [localhost] => (item=/tmp/ansible/files/foo/baz.conf) => {
    "item": "/tmp/ansible/files/foo/baz.conf"
}
ok: [localhost] => (item=/tmp/ansible/files/bar/yatta.conf) => {
    "item": "/tmp/ansible/files/bar/yatta.conf"
}
ok: [localhost] => (item=/tmp/ansible/files/bar/baz.conf) => {
    "item": "/tmp/ansible/files/bar/baz.conf"
}
```

##### ACTUAL RESULTS

```
TASK [debug] ******************************************************************************                                                                                          
ok: [localhost] => (item=/tmp/ansible/files/foo/yatta.conf) => {                                                                                                      
    "item": "/tmp/ansible/files/foo/yatta.conf"                                                                                                                       
}
ok: [localhost] => (item=/tmp/ansible/files/foo/baz.conf) => {
    "item": "/tmp/ansible/files/foo/baz.conf"
}
ok: [localhost] => (item=/tmp/ansible/files/bar/yatta.conf) => {
    "item": "/tmp/ansible/files/bar/yatta.conf"
}
ok: [localhost] => (item=/tmp/ansible/files/bar/baz.conf) => {
    "item": "/tmp/ansible/files/bar/baz.conf"
}
TASK [debug] ******************************************************************************
fatal: [localhost]: FAILED! => {"msg": "An unhandled exception occurred while running the lookup plugin 'fileglob'. Error was a <type 'exceptions.AttributeError'>, original message: 
'list' object has no attribute 'rfind'"}
```

### PR DESCRIPTION
**Pull request by samdoran**
##### SUMMARY

Fixes #48788

##### ISSUE TYPE

- Docs Pull Request

##### COMPONENT NAME

`docs/docsite/rst/user_guide/complex_data_manipulation.rst`

### CODE REVIEW

**Comment by bcoca** [comment]
you have a syntax error

**Comment by samdoran** [comment]
> you have a syntax error

Any hints? I'm not seeing it.

File: docs/docsite/rst/user_guide/complex_data_manipulation.rst [hunk 1]
```diff
@@ -136,6 +136,29 @@ Another way is to avoid adding elements to the list in the first place, so you c
...
+          - prod
+          - web
+      tasks:
+        - name: Copy a glob of files based on a list of groups
+          copy:
+            src: "{{ item }}"
+            dest: "/tmp/{{ item }}"
+          loop: '{{ q("fileglob", *globlist) }}'
+          vars:
+            globlist: '{{ mygroups | map("regex_replace", "^(.*)$", "files/\1/*.conf" | list }}'
```
Comment (sivel) [inline docs/docsite/rst/user_guide/complex_data_manipulation.rst]:
```suggestion
            globlist: '{{ mygroups | map("regex_replace", "^(.*)$", "files/\1/*.conf") | list }}'
```

Comment (bcoca) [reply docs/docsite/rst/user_guide/complex_data_manipulation.rst]: ^ da missing parens ... sorry, thought it was obvious once pointed out ... forget not everyone has Jinja2 parser integrated neural pathways.

Comment (samdoran) [reply docs/docsite/rst/user_guide/complex_data_manipulation.rst]: Gah. Sorry I couldn't find that.

**Review by samccann — APPROVED** [review]

**Pull request merged**